addon/adapters/twitch.js

```diff
@@ -1,35 +1,47 @@
 import JSONAPIAdapter from 'ember-data/adapters/json-api';
 import Inflector from 'ember-inflector';
-import { camelize } from 'ember-string';
+import TwitchConfig from 'ember-data-twitch/configuration';
 
 const { inflector } = Inflector;
 
-
 export default JSONAPIAdapter.extend({
   host: 'https://api.twitch.tv',
   namespace: 'kraken',
-  dataType: 'jsonp',
+  defaultSerializer: '-twitch',
   pluralizePath: true,
 
+  init() {
+    this._super(...arguments);
+
+    if (!TwitchConfig.clientID) {
+      throw new Error(`please set a "clientID" in the ENV['ember-data-twitch'] hash of \`config/environment.js\``);
+    }
+  },
+
+  headers: {
+    Accept: 'application/vnd.twitchtv.v3+json',
+    'Client-ID': TwitchConfig.clientID
```

Comment:
That's a really interesting question. 

I can't think of a reason at the moment, but I also can't think of a reason why not -- we certainly wouldn't lose anything by adding extra level of configurability, and it would enable other potential uses of `clientID` in the future. I went ahead and made the change.